src/pages/List/List.js

```diff
@@ -1,7 +1,223 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import CardList from './Card/CardList';
+import Filter from './Filter/Filter';
+import styled from 'styled-components';
+import { DateObject } from 'react-multi-date-picker';
+
+const DATE_OBJECT = [
+  new DateObject().subtract(0, 'days'),
+  new DateObject().add(0, 'days'),
+];
 
 const List = () => {
-  return <div>List</div>;
+  const [personCount, setPersonCount] = useState(0);
+  const [cardList, setCardList] = useState([]);
+  const [startDatePick, setStartDatePick] = useState('');
+  const [endDatePick, setEndDatePick] = useState('');
+  const [stayType, setStayType] = useState();
+  const [countModalActive, setCountModalActive] = useState(false);
+  const [stayModalActive, setStayModalActive] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [selectLocation, setSelectLocation] = useState('');
+  const [queryArray, setQueryArray] = useState({
+    search: '',
+    person: '',
+    stayType: '',
+    startDatePick: '',
+    endDatePick: '',
+    region: '',
+  });
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [values, setValues] = useState(DATE_OBJECT);
+
+  const urlApi = `?${queryArray.person ? `people=${queryArray.person}` : ''}${
+    queryArray.stayType ? `&category=${queryArray.stayType}` : ''
+  }${queryArray.region ? `&region=${queryArray.region}` : ''}${
+    queryArray.startDatePick
+      ? `&check_in=${queryArray.startDatePick}&check_out=${queryArray.endDatePick}`
+      : ''
+  }${queryArray.search ? `&search=${queryArray.search}` : ''}`;
+
+  useEffect(() => {
+    if (location.state > 0) {
+      setStartDatePick(location.state.startDatePick);
+      setEndDatePick(location.state.endDatePick);
+    }
+  }, [location.state]);
+
+  useEffect(() => {
+    fetch('http://10.58.4.63:8000/residences')
+      .then(res => res.json())
+      .then(data => setCardList(data.residences_list));
+  }, []);
+
+  useEffect(() => {
+    fetch(`http://10.58.4.63:8000/residences${location.search}`)
+      .then(res => res.json())
+      .then(data => setCardList(data.residences_list));
+  }, [location.search]);
+
+  useEffect(() => {
+    navigate(`/list${urlApi}`);
+  }, [navigate, urlApi]);
+
+  useEffect(() => {
+    setQueryArray(prev => ({
+      ...prev,
+      startDatePick: startDatePick,
+      endDatePick: endDatePick,
+    }));
+  }, [startDatePick, endDatePick]);
+
```

Comment:
```suggestion
  useEffect(() => {
    setQueryArray(prev => ({
      ...prev,
      startDatePick,
      endDatePick,
    }));
  }, [startDatePick, endDatePick]);

```

key, value가 같으면 이렇게 줄일 수 있습니다